# **Statistical Analysis and Hypothesis Testing**



## Objectives

- Apply descriptive statistical measures to the cleaned airline passenger satisfaction dataset.
- Test selected hypotheses related to passenger satisfaction.
- Use appropriate statistical tests based on the type of variables being analysed.
- Interpret p-values and statistical significance in the context of the business problem.
- Summarise the results in clear business-focused conclusions.

## Inputs

- `data/clean_data/airline_clean.csv`

## Outputs

- Descriptive statistical summaries.
- Results of hypothesis tests.
- Interpretation of statistical significance.
- Evidence to support or reject the project hypotheses.



---

# Change working directory

In [1]:
import os

current_dir = os.getcwd()
current_dir

'c:\\Users\\hanee\\OneDrive\\Desktop\\Project2\\airline_passenger_satisfaction_analyst\\airline-passenger-satisfaction\\jupyter_notebooks'

In [2]:
if os.path.basename(os.getcwd()) == "airline_passenger_satisfaction_analyst":
    os.chdir("airline-passenger-satisfaction")
elif os.path.basename(os.getcwd()) == "jupyter_notebooks":
    os.chdir(os.path.dirname(os.getcwd()))

print(f"Current working directory: {os.getcwd()}")

Current working directory: c:\Users\hanee\OneDrive\Desktop\Project2\airline_passenger_satisfaction_analyst\airline-passenger-satisfaction


Confirm the new current directory

In [3]:
current_dir = os.getcwd()
current_dir

'c:\\Users\\hanee\\OneDrive\\Desktop\\Project2\\airline_passenger_satisfaction_analyst\\airline-passenger-satisfaction'

## Load the Cleaned Dataset

Load the cleaned airline passenger satisfaction dataset and import the libraries required for statistical analysis.

In [4]:
import pandas as pd
from scipy import stats

df = pd.read_csv("data/clean_data/airline_clean.csv")

df.head()

,Gender,Customer Type,Age,Type of Travel,Class,Flight Distance,Inflight wifi service,Departure/Arrival time convenient,Ease of Online booking,Gate location,...,Inflight entertainment,On-board service,Leg room service,Baggage handling,Checkin service,Inflight service,Cleanliness,Departure Delay in Minutes,Arrival Delay in Minutes,satisfaction
0,Male,Loyal Customer,13,Personal Travel,Eco Plus,460,3,4,3,1,...,5,4,3,4,4,5,5,25,18.0,neutral or dissatisfied
1,Male,disloyal Customer,25,Business travel,Business,235,3,2,3,3,...,1,1,5,3,1,4,1,1,6.0,neutral or dissatisfied
2,Female,Loyal Customer,26,Business travel,Business,1142,2,2,2,2,...,5,4,3,4,4,4,5,0,0.0,satisfied
3,Female,Loyal Customer,25,Business travel,Business,562,2,5,5,5,...,2,2,5,3,1,4,2,11,9.0,neutral or dissatisfied
4,Male,Loyal Customer,61,Business travel,Business,214,3,3,3,3,...,3,3,4,4,3,3,3,0,0.0,satisfied


## Descriptive Statistics

Summarise selected numerical variables using measures of central tendency and spread before carrying out hypothesis testing.

In [5]:
statistical_columns = [
    "Age",
    "Flight Distance",
    "Online boarding",
    "Inflight entertainment",
    "Seat comfort",
    "Departure Delay in Minutes",
    "Arrival Delay in Minutes"
]

df[statistical_columns].describe().round(2)

,Age,Flight Distance,Online boarding,Inflight entertainment,Seat comfort,Departure Delay in Minutes,Arrival Delay in Minutes
count,103594.00,103594.00,103594.00,103594.00,103594.00,103594.00,103594.00
mean,39.38,1189.33,3.25,3.36,3.44,14.75,15.18
std,15.11,997.30,1.35,1.33,1.32,38.12,38.70
min,7.00,31.00,0.00,0.00,0.00,0.00,0.00
25%,27.00,414.00,2.00,2.00,2.00,0.00,0.00
50%,40.00,842.00,3.00,4.00,4.00,0.00,0.00
75%,51.00,1743.00,4.00,4.00,5.00,12.00,13.00
max,85.00,4983.00,5.00,5.00,5.00,1592.00,1584.00


### Descriptive Statistics Finding

The descriptive statistics provide an overview of the central tendency and spread of the selected numerical variables.

Passenger age has a mean of 39.38 years and a median of 40 years, indicating that the age distribution is relatively balanced around its centre. Flight distance has a mean of 1,189.33 and a median of 842, suggesting a positively skewed distribution.

The service-rating variables generally have median ratings between 3 and 4 on the 0–5 scale.

Departure and arrival delays show a substantial difference between their means and medians. The mean departure delay is 14.75 minutes and the mean arrival delay is 15.18 minutes, while both have a median of 0 minutes. Their maximum values exceed 1,500 minutes, indicating strongly right-skewed distributions influenced by a relatively small number of very large delays.

### Variance

Variance measures how widely observations are spread around the mean. A higher variance indicates greater variability in the observations.

For this project, variance is calculated for departure and arrival delays to quantify the variability in passenger flight delays.

In [6]:
delay_variance = df[
    [
        "Departure Delay in Minutes",
        "Arrival Delay in Minutes"
    ]
].var()

delay_variance

Departure Delay in Minutes    1452.885641
Arrival Delay in Minutes      1497.587990
dtype: float64

## Hypothesis Testing

### Hypothesis Test 1 — Type of Travel and Passenger Satisfaction

During exploratory analysis, satisfaction levels differed noticeably between business and personal travel passengers. A Chi-square test of independence is used to determine whether this observed relationship is statistically significant.

**Null Hypothesis (H₀):** Type of travel and passenger satisfaction are independent; there is no statistically significant association between them.

**Alternative Hypothesis (H₁):** Type of travel and passenger satisfaction are associated; there is a statistically significant relationship between them.

**Significance level:** α = 0.05

A Chi-square test of independence is appropriate because both `Type of Travel` and `satisfaction` are categorical variables.

#### Observed Frequencies

Create a contingency table showing the number of satisfied and neutral or dissatisfied passengers within each type of travel.

In [7]:
travel_contingency = pd.crosstab(
    df["Type of Travel"],
    df["satisfaction"]
)

travel_contingency

satisfaction,neutral or dissatisfied,satisfied
Type of Travel,,
Business travel,29831,41634
Personal Travel,28866,3263


#### Chi-square Test of Independence

Calculate the Chi-square statistic and p-value to determine whether the association between type of travel and passenger satisfaction is statistically significant.

In [8]:
chi2, p_value, dof, expected = stats.chi2_contingency(
    travel_contingency
)

print(f"Chi-square statistic: {chi2:.2f}")
print(f"P-value: {p_value:.6f}")
print(f"Degrees of freedom: {dof}")
print(f"Minimum expected frequency: {expected.min():.2f}")

Chi-square statistic: 20882.22
P-value: 0.000000
Degrees of freedom: 1
Minimum expected frequency: 13924.51


#### Result and Interpretation

The Chi-square test produced a statistic of 20,882.22 with 1 degree of freedom. The p-value is less than 0.001, which is below the significance level of 0.05.

The minimum expected frequency is 13,924.51 which is well above 5, indicating that the expected-frequency assumption for the Chi-square test is satisfied.

Therefore, the **null hypothesis (H₀) is rejected**.

There is statistically significant evidence of an association between **type of travel and passenger satisfaction**. The observed frequencies show that business travellers are considerably more likely to be satisfied than personal travellers.

This supports the pattern identified during exploratory data analysis and suggests that type of travel is an important variable to consider when analysing and later predicting passenger satisfaction.

---

#### Effect Size — Cramér's V

Calculate Cramér's V to measure the strength of the association between type of travel and passenger satisfaction.

In [9]:
import numpy as np

n = travel_contingency.to_numpy().sum()

cramers_v = np.sqrt(
    chi2 / (
        n * min(
            travel_contingency.shape[0] - 1,
            travel_contingency.shape[1] - 1
        )
    )
)

print(f"Cramér's V: {cramers_v:.3f}")

Cramér's V: 0.449


#### Effect Size Interpretation

Cramér's V is 0.449, indicating a moderate to relatively strong association between "type of travel and passenger satisfaction".

This supports the Chi-square test result by showing that the relationship is not only statistically significant but also meaningful in practical terms.

Therefore, type of travel appears to be an important characteristic associated with passenger satisfaction and may provide useful predictive information in later modelling.

---

## Core Statistical Concepts

### Mean

The mean is the average value of a numerical variable. It is calculated by adding all values and dividing by the number of observations. In this project, the mean is useful for understanding typical values such as passenger age, flight distance and flight delays. However, the mean can be influenced by unusually high or low values.

### Median

The median is the middle value when observations are arranged in order. It is less affected by extreme values than the mean. This is particularly useful for the flight delay variables, where the median delay is 0 minutes while the mean is higher because some passengers experience very large delays.

### Standard Deviation

Standard deviation measures how spread out observations are around the mean. A larger standard deviation indicates greater variability, while a smaller standard deviation indicates that observations are more concentrated around the mean.

### Hypothesis Testing

Hypothesis testing is used to determine whether evidence from the data is strong enough to reject a null hypothesis. In this project, a significance level of **0.05** is used. A p-value below 0.05 provides evidence to reject the null hypothesis.

### Basic Probability

Probability describes the likelihood of an event occurring and ranges from 0 to 1. In this dataset, the probability of selecting a satisfied passenger can be estimated by dividing the number of satisfied passengers by the total number of passengers.

---

### Probability of Passenger Satisfaction

Calculate the probability that a randomly selected passenger from the dataset is satisfied.

In [10]:
satisfied_count = (df["satisfaction"] == "satisfied").sum()
total_passengers = len(df)

satisfaction_probability = satisfied_count / total_passengers

print(f"Satisfied passengers: {satisfied_count}")
print(f"Total passengers: {total_passengers}")
print(f"Probability of satisfaction: {satisfaction_probability:.4f}")
print(f"Percentage satisfied: {satisfaction_probability * 100:.2f}%")

Satisfied passengers: 44897
Total passengers: 103594
Probability of satisfaction: 0.4334
Percentage satisfied: 43.34%


### Probability Interpretation

The probability of randomly selecting a satisfied passenger from the dataset is approximately 0.433, or 43.3%.

This means that fewer than half of the passengers in the cleaned dataset are satisfied, which is consistent with the satisfaction distribution identified during exploratory data analysis.

## Conclusions

The statistical analysis provided further evidence for patterns identified during exploratory data analysis.

Descriptive statistics demonstrated the use of the "mean, median and standard deviation" to understand the centre and spread of numerical variables. The delay variables showed an important difference between their mean and median values, indicating that unusually large delays influence their distributions.

Basic probability analysis showed that the probability of randomly selecting a satisfied passenger is approximately "43.3%", meaning that fewer than half of the passengers in the dataset are satisfied.

A Chi-square test of independence was used to investigate the relationship between "type of travel and passenger satisfaction". The test produced a p-value below "0.05", providing sufficient evidence to reject the null hypothesis and conclude that there is a statistically significant association between type of travel and passenger satisfaction.

Cramér's V was "0.449", indicating that the association is meaningful in strength and not simply statistically significant because of the large dataset.

Overall, the statistical findings support the exploratory analysis and indicate that passenger satisfaction is meaningfully associated with travel characteristics. These findings provide evidence for selecting relevant features during the machine learning stage of the project.

## Next Steps

The next stage of the project will apply machine learning techniques to predict passenger satisfaction using passenger characteristics, travel information and service experience variables.